apis/database/v1beta1/nosql_types.go

```diff
@@ -0,0 +1,164 @@
+	// DatabaseAccountOfferType - The offer type for the database
+	DatabaseAccountOfferType string `json:"databaseAccountOfferType"`
+	// IPRangeFilter - Cosmos DB Firewall Support: This value specifies the set of IP addresses or IP address ranges in CIDR form to be included as the allowed list of client IPs for a given database account. IP addresses/ranges must be comma separated and must not contain any spaces.
+	IPRangeFilter *string `json:"ipRangeFilter,omitempty"`
+	// EnableAutomaticFailover - Enables automatic failover of the write region in the rare event that the region is unavailable due to an outage. Automatic failover will result in a new write region for the account and is chosen based on the failover priorities configured for the account.
+	EnableAutomaticFailover *bool `json:"enableAutomaticFailover,omitempty"`
+	// EnableMultipleWriteLocations - Enables the account to write in multiple locations
+	EnableMultipleWriteLocations *bool `json:"enableMultipleWriteLocations,omitempty"`
+	// EnableCassandraConnector - Enables the cassandra connector on the Cosmos DB C* account
+	EnableCassandraConnector *bool `json:"enableCassandraConnector,omitempty"`
+}
+
+// CosmosDBAccountConsistencyPolicy the consistency policy for the Cosmos DB database account.
+type CosmosDBAccountConsistencyPolicy struct {
+	// DefaultConsistencyLevel - The default consistency level and configuration settings of the Cosmos DB account. Possible values include: 'Eventual', 'Session', 'BoundedStaleness', 'Strong', 'ConsistentPrefix'
+	DefaultConsistencyLevel string `json:"defaultConsistencyLevel"`
+	// MaxStalenessPrefix - When used with the Bounded Staleness consistency level, this value represents the number of stale requests tolerated. Accepted range for this value is 1 – 2,147,483,647. Required when defaultConsistencyPolicy is set to 'BoundedStaleness'.
+	MaxStalenessPrefix *int64 `json:"maxStalenessPrefix,omitempty"`
+	// MaxIntervalInSeconds - When used with the Bounded Staleness consistency level, this value represents the time amount of staleness (in seconds) tolerated. Accepted range for this value is 5 - 86400. Required when defaultConsistencyPolicy is set to 'BoundedStaleness'.
+	MaxIntervalInSeconds *int32 `json:"maxIntervalInSeconds,omitempty"`
+}
+
+// CosmosDBAccountLocation a region in which the Azure Cosmos DB database account is deployed.
+type CosmosDBAccountLocation struct {
+	// LocationName - The name of the region.
+	LocationName string `json:"locationName"`
+	// FailoverPriority - The failover priority of the region. A failover priority of 0 indicates a write region. The maximum value for a failover priority = (total number of regions - 1). Failover priority values must be unique for each of the regions in which the database account exists.
+	FailoverPriority int32 `json:"failoverPriority"`
+	// IsZoneRedundant - Flag to indicate whether or not this region is an AvailabilityZone region
+	IsZoneRedundant bool `json:"isZoneRedundant"`
+}
```

Comment:
Nit: Some of the comments in this file are tough to follow without word wraps. I use https://marketplace.visualstudio.com/items?itemName=stkb.rewrap to automatically wrap comments at 80 chars.